lib/ansible/plugins/list.py

```diff
@@ -116,7 +123,6 @@ def list_collection_plugins(ptype, collections, search_paths=None):
 
     # starts at  {plugin_name: filepath, ...}, but changes at the end
     plugins = {}
-    dirs = []
```

Comment:
keep as if collections is empty, dirs won't be defined for 146/152

Reply:
It's defined in the `if`, `elif`, and `else` and then used in `_list_plugins_from_paths` for each collection in the loop, so it is always be defined. This was to prevent searching irrelevant dirs multiple times when there are multiple collections.

It does look like there's a bug listing all collections when collections is empty (because list_collections isn't getting the playbook adj search path). I will fix that and add a test (tomorrow).